pkgs/by-name/co/comfyui/package.nix

```
@@ -0,0 +1,103 @@
+{ lib
+, python311
+, linkFarm
+, writers
+, writeTextFile
+, fetchFromGitHub
+, stdenv
+, torch ? python311.pkgs.torch
+, modelsPath ? "/var/lib/comfyui/models"
+, inputPath ? "/var/lib/comfyui/input"
+, outputPath ? "/var/lib/comfyui/output"
+, customNodes ? []
+}:
+
+let
+  modelPathsFile = writeTextFile {
+    name = "model_paths.yaml";
+    text = (lib.generators.toYAML {} {
+      comfyui = {
+        base_path = modelsPath;
+        checkpoints = "${modelsPath}/checkpoints";
+        clip = "${modelsPath}/clip";
+        clip_vision = "${modelsPath}/clip_vision";
+        configs = "${modelsPath}/configs";
+        controlnet = "${modelsPath}/controlnet";
+        embeddings = "${modelsPath}/embeddings";
+        loras = "${modelsPath}/loras";
+        upscale_models= "${modelsPath}/upscale_models";
+        vae = "${modelsPath}/vae";
+        vae_approx = "${modelsPath}/vae_approx";
+      };
+    });
+  };
+
+  pythonEnv = (python311.withPackages (ps: [ torch ] ++ [
+    ps.torchsde
+    ps.torchvision
+    ps.torchaudio
+    ps.transformers
+    ps.safetensors
+    ps.accelerate
+    ps.aiohttp
+    ps.einops
+    ps.pyyaml
+    ps.pillow
+    ps.scipy
+    ps.psutil
+    ps.tqdm
```

```suggestion
  pythonEnv = (python3Packages.python.withPackages (ps: with ps; [
    (
      if gpuBackend == "cuda"
      then torchWithCuda
      else if gpuBackend == "rocm"
      then torchWithRocm
      else torch
    )
    torchsde
    torchvision
    torchaudio
    transformers
    safetensors
    accelerate
    aiohttp
    einops
    pyyaml
    pillow
    scipy
    psutil
    tqdm
```

Many of the torch-related packages here likely need rocm variants.

---

yes, off the top of my head I think `torchsde`, `torchvision`, `torchaudio`, `transformers`, `safetensors`, `accelerate` all depend on `torch` and therefore they need some kind of rocm variant.

I have tried doing an override on them here, to pass the correct `torch` reference, but it didn't seem like the right way to do it.

I am thinking that we either need to declare a variant for each one like we do for torch, or I will need to do an override here and pass the appropriate `gpuBackend`. However I am not sure which one I should do